projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php

```diff
@@ -42,16 +40,63 @@ public function register_routes() {
 					'methods'             => WP_REST_Server::EDITABLE,
 					'callback'            => array( $this, 'update_site_options' ),
 					'permission_callback' => array( $this, 'can_access' ),
-					'request_format'      => array(
-						'checklist_statuses' => '(array) Array of launchpad checklist tasks completion status',
+					'args'                => array(
+						'checklist_statuses' => array(
+							'description' => 'Launchpad statuses',
+							'type'        => 'array',
+							'items'       => array(
+								'domain_upsell_deferred' => array(
+									'type' => 'boolean',
+								),
+								'links_edited'           => array(
+									'type' => 'boolean',
+								),
+								'site_edited'            => array(
+									'type' => 'boolean',
+								),
+								'site_launched'          => array(
+									'type' => 'boolean',
+								),
+								'first_post_published'   => array(
+									'type' => 'boolean',
+								),
+								'video_uploaded'         => array(
+									'type' => 'boolean',
+								),
+								'publish_first_course'   => array(
+									'type' => 'boolean',
+								),
+								'plan_completed'         => array(
+									'type' => 'boolean',
+								),
+							),
+						),
+						'site_intent'        => array(
```

Comment:
This is thorough and comprehensive. Thank you for enhancing this!

We currently haven't had a need for `site_intent` and `launchpad_screen` here since these generally get handled with other mechanisms in site creation, or on wp hooks on the backend (like for turning off launchpad_screen). And with their simplicity have been easy to use with existing site options endpoints and mechanisms. But bringing launchpad related options into this endpoint makes sense to me, and seems forward looking and like good consolidation.

`launchpad_screen` makes sense here definitely. But im wondering if `site_intent` belongs here? While we currently use it to determine the onboarding checklist to show, its a bit separate from the launchpad itself and we do not update its value based on launchpad interactions.

Reply:
> But im wondering if site_intent belongs here? While we currently use it to determine the onboarding checklist to show, its a bit separate from the launchpad itself and we do not update its value based on launchpad interactions.

I agree with this sentiment 👍